stock_component_account/models/product_product.py

```diff
@@ -0,0 +1,35 @@
+# © 2020 - today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import models
+
+
+class Product(models.Model):
+
+    _inherit = "product.product"
+
+    def _compute_average_price(self, qty_done, quantity, moves):
+        is_serialized_move = bool(moves.mapped("move_line_ids.lot_id.is_serial"))
+        if is_serialized_move:
+            return self._compute_average_price_including_components(
+                qty_done, quantity, moves
+            )
+        else:
+            return super()._compute_average_price(qty_done, quantity, moves)
+
+    def _compute_average_price_including_components(self, qty_done, quantity, moves):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/numigi/odoo-stock-addons/pullRequest?prid=5314676)